src/huggingface_hub/cli/cache.py

```diff
@@ -54,122 +62,93 @@ def _inner(*args, **kwargs):
-        cache_subparsers = cache_parser.add_subparsers(dest="cache_command", help="Cache subcommands")
+cache_app = typer.Typer(help="Manage local cache directory.", rich_markup_mode=None)
 
-        # Show help if no subcommand is provided
-        cache_parser.set_defaults(func=lambda args: cache_parser.print_help())
 
-        # Scan subcommand
-        scan_parser = cache_subparsers.add_parser("scan", help="Scan cache directory.")
-        scan_parser.add_argument(
-            "--dir",
-            type=str,
-            default=None,
-            help="cache directory to scan (optional). Default to the default HuggingFace cache.",
-        )
-        scan_parser.add_argument(
-            "-v",
-            "--verbose",
-            action="count",
-            default=0,
-            help="show a more verbose output",
-        )
-        scan_parser.set_defaults(func=CacheCommand, cache_command="scan")
-        # Delete subcommand
-        delete_parser = cache_subparsers.add_parser("delete", help="Delete revisions from the cache directory.")
-        delete_parser.add_argument(
-            "--dir",
-            type=str,
-            default=None,
-            help="cache directory (optional). Default to the default HuggingFace cache.",
-        )
-        delete_parser.add_argument(
-            "--disable-tui",
-            action="store_true",
-            help=(
-                "Disable Terminal User Interface (TUI) mode. Useful if your platform/terminal doesn't support the multiselect menu."
-            ),
-        )
-        delete_parser.add_argument(
-            "--sort",
-            nargs="?",
-            choices=["alphabetical", "lastUpdated", "lastUsed", "size"],
-            help=(
-                "Sort repositories by the specified criteria. Options: "
-                "'alphabetical' (A-Z), "
-                "'lastUpdated' (newest first), "
-                "'lastUsed' (most recent first), "
-                "'size' (largest first)."
-            ),
-        )
-        delete_parser.set_defaults(func=CacheCommand, cache_command="delete")
-
-    def __init__(self, args: Namespace) -> None:
-        self.args = args
-        self.verbosity: int = getattr(args, "verbose", 0)
-        self.cache_dir: Optional[str] = getattr(args, "dir", None)
-        self.disable_tui: bool = getattr(args, "disable_tui", False)
-        self.sort_by: Optional[SortingOption_T] = getattr(args, "sort", None)
-        self.cache_command: Optional[str] = getattr(args, "cache_command", None)
-
-    def run(self):
-        if self.cache_command == "scan":
-            self._run_scan()
-        elif self.cache_command == "delete":
-            self._run_delete()
+@cache_app.command("scan", help="Scan the cache directory")
+def cache_scan(
+    dir: Optional[str] = typer.Option(
+        None,
+        "--dir",
+        help="Cache directory to scan (defaults to Hugging Face cache).",
+    ),
```

Comment:
```suggestion
    dir: Annotated[Optional[str], typer.Option(help="Cache directory to scan (defaults to Hugging Face cache).")] = None
```

from what I understand in https://typer.tiangolo.com/tutorial/options/help/, the recommended way to use Typer is to define options with an `Annotated` annotation.

<img width="789" height="237" alt="image" src="https://github.com/user-attachments/assets/bfea7f36-3087-4d83-8017-497bb2c331a9" />


Also, providing `--dir` for the `dir` argument is redundant (and therefore can be avoided)

(same comment on all commands)

Reply:
yes! addressed in https://github.com/huggingface/huggingface_hub/pull/3365/commits/4437b4c860d322d16fc53c4ed50a9660693a850d